lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -61,9 +61,24 @@ func (d *zfs) CreateVolume(vol Volume, filler *VolumeFiller, op *operations.Oper
 		if exists {
 			canRestore := true
 
-			// For block volumes check if the cached image volume is larger than the current pool volume.size
-			// setting (if so we won't be able to resize the snapshot to that the smaller size later).
-			if vol.contentType == ContentTypeBlock {
+			zfsVolType, err := d.getDatasetProperty(d.dataset(vol, true), "type")
```

Comment:
This isn't a blocker, but I still think it would be cheaper to just test for a non-empty `block.filesystem` setting on the image `vol`'s config here rather than calling out to the ZFS layer.

Reply:
Did you figure out why we can't use the volume config here yet?

Reply:
@monstermunchkin any news on this?

Reply:
@monstermunchkin now that we are creating the volume DB record before creating the image ZFS volume this shouldn't be needed and we can decide the volume type from the volume's config.